src/validation.ts

```diff
@@ -138,12 +139,7 @@ export function validateSettingsTCP(settings: TcpSettings): boolean {
 }
 
 function validateHttpTarget(target: string): boolean {
-  try {
-    const url = new URL(target);
-    return url.protocol === 'https:' || url.protocol === 'http:';
-  } catch (_) {
-    return false;
-  }
+  return Boolean(validUrl.isWebUri(target));
```

Comment:
we should add a test for this

Reply:
We have a fair few tests right now for URL validation https://github.com/grafana/synthetic-monitoring-app/blob/215302c5c67df6e03c8ec8b17a17a414855490a6/src/validation.test.ts#L130

For the specific case you raised, it validates as a good URL, so I didn't add a case for it. Happy to add more cases, but I'm not sure what isn't covered

Reply:
http://hostname/ (no domain) should be added to bad targets, because even if it's a valid URL, it's not a URL that we can work with (as probes have no way of resolving that).

Reply:
Yeah, good point. I'll open another issue to follow up on this